app/src/main/java/org/fossasia/openevent/api/processor/VersionApiProcessor.java

```diff
@@ -104,7 +107,7 @@ public void run() {
 
     @Override
     public void failure(RetrofitError error) {
-        OpenEventApp.getEventBus().post(new CantDownloadEvent());
```

Comment:
I think you should not depend on APIClient for your implementation. The code should be as decoupled as possible. You should continue firing the error or at least log it, but whether the MainActivity chooses to subscribe to it or whether it has shown a network error dialog, should be upto it.

Also I think that instead of setting an Error Handler on the Retrofit adapter you should piggy back on the event that the processor generates and transform the event into something that you can show (the dialog in your case). The reason I am saying this is because the error generated from the Processor is local, has more contextual information, you know exactly where the error is coming from, while for the Retrofit adapter it will cause the handler to fire for all global error issues. I don't think that should be the approach.